app/storage/blob-storage.js

```diff
@@ -0,0 +1,55 @@
+const { DefaultAzureCredential } = require('@azure/identity')
+const { BlobServiceClient } = require('@azure/storage-blob')
+const { storageConfig } = require('../config')
+
+let blobServiceClient
+let containersInitialised
+
+if (storageConfig.useConnectionStr) {
+  console.log('Using connection string for BlobServiceClient')
+  blobServiceClient = BlobServiceClient.fromConnectionString(storageConfig.connectionStr)
+} else {
+  console.log('Using DefaultAzureCredential for BlobServiceClient')
+  const credential = new DefaultAzureCredential({ managedIdentityClientId: storageConfig.managedIdentityClientId })
+  blobServiceClient = new BlobServiceClient(storageConfig.endpoint, credential)
+}
+
+const container = blobServiceClient.getContainerClient(storageConfig.container)
+
+const initialiseFolders = async () => {
+  const placeHolderText = 'Placeholder'
+  const client = container.getBlockBlobClient(`${storageConfig.folder}/default.txt`)
+  await client.upload(placeHolderText, placeHolderText.length)
+}
+
+const initialiseContainers = async () => {
+  if (storageConfig.createContainers) {
+    console.log('Making sure blob containers exist')
+    await container.createIfNotExists()
+  }
+  await initialiseFolders()
+  containersInitialised = true
+}
+
+const getOutboundBlobClient = async (filename) => {
+  if (!containersInitialised) {
+    await initialiseContainers()
+  }
+  return container.getBlockBlobClient(`${storageConfig.folder}/${filename}`)
+}
+
+const createAzuriteInfrastructure = async () => {
+  if (process.env.NODE_ENV === 'development') {
+    await initialiseContainers()
```

Comment:
I think the logic should just be on the log message otherwise when in deployment environment we will not initialise containers which means we wont actually be checking they are available? Do you agree?

You could just change the log message to be generic eg `storage account ready to receive files` which works for both local and deployed